### PR TITLE
[web_benchmarks] Migrate to PlatformDispatcher API

### PR DESCRIPTION
**Pull request by goderbauer**
Part of https://github.com/flutter/flutter/issues/116929.

Migrates away from the soon to be deprecated `window` singleton API to the PlatformDispatcher API.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/web_benchmarks/lib/src/recorder.dart [hunk 1]
```diff
@@ -238,7 +238,7 @@ abstract class SceneBuilderRecorder extends Recorder {
     final Completer<Profile> profileCompleter = Completer<Profile>();
     _profile = Profile(name: name);
 
-    window.onBeginFrame = (_) {
+    PlatformDispatcher.instance.onBeginFrame = (_) {
```
Comment (goderbauer) [inline packages/web_benchmarks/lib/src/recorder.dart:241]: This API was introduced almost three years ago (https://github.com/flutter/engine/pull/20496) and has been on the stable channel for a long time. Migrating to it directly will not scarify backwards compatibility to any of the versions we care about.

File: packages/web_benchmarks/lib/src/recorder.dart [hunk 2]
```diff
@@ -247,22 +247,23 @@ abstract class SceneBuilderRecorder extends Recorder {
...
       try {
         _profile.record('drawFrameDuration', () {
           final SceneBuilder sceneBuilder = SceneBuilder();
           onDrawFrame(sceneBuilder);
           _profile.record('sceneBuildDuration', () {
             final Scene scene = sceneBuilder.build();
             _profile.record('windowRenderDuration', () {
-              window.render(scene);
+              // TODO(goderbauer): Migrate to PlatformDispatcher.implicitView once v3.9.0 is the oldest supported Flutter version.
+              window.render(scene); // ignore: deprecated_member_use
```
Comment (goderbauer) [inline packages/web_benchmarks/lib/src/recorder.dart:259]: The replacement API for this particular call was only introduced recently, so we cannot migrate to it yet to remain backwards compatible with older versions. Ignoring the deprecation warning for the upcoming deprecation in https://github.com/flutter/engine/pull/39302.

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**